### ISSUE TITLE
App cannot pull content from Invidious

### ISSUE THREAD

**Issue by ghost** [stephane-r/Youtube-Audio-Player#169]
Good morning,
Whenever I start the HoloPlay app (version 1.12.1) it fails to pull content from Invidious and prompts me to switch the instance. However, none of them are down (they work in my browser) and all of them fail in HoloPlay.

I am using a FairPhone 4, 256GB, with Android 11 and December 5th security updates.
I have the same issue on a Google Pixel 3 with Android 12.
I did try using a different network with no results.
I don't use a DNS blocker or anything like that on network or phone level. I tried monitoring the connection to see if it gets blocked for whatever reason but the app seems to ping Invidious totally fine:
![Screenshot_20220202-233005_1](https://user-images.githubusercontent.com/76809077/152249228-e84702c4-cb0c-42e4-bca3-efbb53d7c446.png)

It IS able to download the list of instances. It is not able to pull any content from any of them.

**Comment by raffaelemancuso** [comment]
Same issue here.

I get:

"No result. Maybe instance is down? Try to change Invidious instance from settings screen".

The instance yewtu.be is perfectly fine and works on the web browser on the same phone in which I'm using HoloPlay.

**Comment by stephane-r** [comment]
Hi guys !

The frontend seems to work, however the API returns an error (example with yewtu.be : https://yewtu.be/api/v1/popular)

I'm currently using [vid.puffyan.us](https://vid.puffyan.us/api/v1/popular), API work nice :)

**Comment by ghost** [comment]
> I'm currently using [vid.puffyan.us](https://vid.puffyan.us/api/v1/popular), API work nice :)

You're right! Maybe then it would be a good idea to check if the API is working before displaying the instance to the user?
I thought the app is broken, while it seems like the Invidious instances I've tried are simply misconfigured :P

**Comment by stephane-r** [comment]
> You're right! Maybe then it would be a good idea to check if the API is working before displaying the instance to the user? I thought the app is broken, while it seems like the Invidious instances I've tried are simply misconfigured :P

Yes ! Good idea, i'll add this soon on the Invidious Instance screen :) Thank you

**Comment by raffaelemancuso** [comment]
> > You're right! Maybe then it would be a good idea to check if the API is working before displaying the instance to the user? I thought the app is broken, while it seems like the Invidious instances I've tried are simply misconfigured :P
> 
> Yes ! Good idea, i'll add this soon on the Invidious Instance screen :) Thank you

Can we left it open to track the change proposed by @stephane-r ?

Also, is there a way to avoid that the Invidious API instance is down?  
I noticed a "Too many requests".  
Can we propose for example login credentials to call the API, as to avoid spam?

**Comment by stephane-r** [comment]
Yes no problem for stay issue opened.

For Invidious API, you have to see directly on the repo of the project 😄 Have you try to call API with Token ?

**Comment by raffaelemancuso** [comment]
> Yes no problem for stay issue opened.
> 
> For Invidious API, you have to see directly on the repo of the project smile Have you try to call API with Token ?

No I haven't.

How do I do that?

**Comment by stephane-r** [comment]
You have the step on the HoloPlay [README](https://github.com/stephane-r/HoloPlay#about-invidious) :)

First, create account on Invidious Instance, ofc. The, generate token (replace the domaine by your) : **invidious.snopyta.org/authorize_token?scopes=:***

Add it on HoloPlay from Settings screen.

**Comment by raffaelemancuso** [comment]
> You have the step on the HoloPlay [README](https://github.com/stephane-r/HoloPlay#about-invidious) :)
> 
> First, create account on Invidious Instance, ofc. The, generate token (replace the domaine by your) : **invidious.snopyta.org/authorize_token?scopes=:***
> 
> Add it on HoloPlay from Settings screen.

Uhm it says "JSON Parse error: Unexpected token: <"

**Comment by DWW256** [comment]
I'm noticing a similar error, where it throws out "JSON Parse error: Unexpected token: e" when I try to add a new token.
